spec/intl.html

```diff
@@ -0,0 +1,372 @@
+
+      <emu-alg>
+        1. Let _dtf_ be _F_.[[DateTimeFormat]].
+        1. Assert: Type(_dtf_) is Object and _dtf_ has an [[InitializedDateTimeFormat]] internal slot.
+        1. <del>If _date_ is not provided or is *undefined*, then</del>
+          1. <del>Let _x_ be Call(%Date_now%, *undefined*).</del>
+        1. <del>Else,</del>
+          1. <del>Let _x_ be ? ToNumber(_date_).</del>
+        1. Return FormatDateTime(_dtf_, <del>_x_</del><ins>_date_</ins>).
+      </emu-alg>
+
+      <p>
+        The `length` property of a DateTime format function is 1.
+      </p>
+    </emu-clause>
+
+    <emu-clause id="sec-partitiondatetimepattern" aoid="PartitionDateTimePattern">
+      <h1>PartitionDateTimePattern ( _dateTimeFormat_, <del>_x_</del><ins>_date_</del> )</h1>
+
+      <p>
+        The PartitionDateTimePattern abstract operation is called with arguments _dateTimeFormat_ (which must be an object initialized as a DateTimeFormat) and <del>_x_</del><ins>_date_</del> (which must be <del>a Number</del><ins>an ECMAScript</ins> value), <del>interprets _x_ as a time value as specified in ES2015, <emu-xref href="#sec-time-values-and-time-range"></emu-xref>,</del> and creates the corresponding parts according to the effective locale and the formatting options of _dateTimeFormat_. The following steps are taken:
+      </p>
+
+      <emu-alg>
+        1. <ins>If IsTemporalObject(_date_) is *true*,</ins>
+          1. <ins>Let _x_ be _date_.</ins>
+          1. <ins>If _date_ is a CivilDate object, then</ins>
+            1. <ins>Let _pattern_ be _dtf_.[[CivilDatePattern]].</ins>
+            1. <ins>Let _tm_ be { [[weekday]]: _date_.dayOfWeek, [[year]]: _date_.year, [[month]]: _date_.month, [[day]]: _date_.day }.</ins>
+          1. <ins>Else if _date_ is a CivilTime object, then</ins>
+            1. <ins>Let _pattern_ be _dtf_.[[CivilTimePattern]].</ins>
+            1. <ins>Let _tm_ be { [[hour]]: _date_.hour, [[minute]]: _date_.minute, [[second]]: _date_.second }.</ins>
+          1. <ins>Else if _date_ is a CivilDateTime object, then</ins>
+            1. <ins>Let _pattern_ be _dtf_.[[CivilDateTimePattern]].</ins>
+            1. <ins>Let _tm_ be { [[weekday]]: _date_.dayOfWeek, [[year]]: _date_.year, [[month]]: _date_.month, [[day]]: _date_.day, [[hour]]: _date_.hour, [[minute]]: _date_.minute, [[second]]: _date_.second }.</ins>
+          1. <ins>Else if _date_ is a ZonedDateTime object, then</ins>
+            1. <ins>Let _pattern_ be _dtf_.[[ZonedDateTimePattern]].</ins>
+            1. Let _offsetDateTime_ be _date_.getOffsetDateTime().
+            1. <ins>Let _tm_ be { [[weekday]]: _offsetDateTime_.dayOfWeek, [[year]]: _offsetDateTime_.year, [[month]]: _offsetDateTime_.month, [[day]]: _offsetDateTime_.day, [[hour]]: _offsetDateTime_.hour, [[minute]]: _offsetDateTime_.minute, [[second]]: _offsetDateTime_.second }.</ins>
+          1. <ins>If _pattern_ is *null*, throw a *TypeError* exception.</ins>
```

Comment:
I support throwing an exception here, but Dan mentioned that maybe the other approach would be to use a fallback for _pattern_ that includes all of the fields if none of the fields for a particular Temporal type were specified.  That would probably be logic for InitializeDateTimeFormat.  Keep it as-is for now, and we should follow up later on exactly which behavior we want here.